mchf-eclipse/drivers/audio/cw/cw_gen.c

```diff
@@ -226,7 +226,7 @@ static const float sm_table[CW_SMOOTH_TBL_SIZE] =
 
 void CwGen_SetSpeed()
 {
-    ps.dit_time         = 1650/ts.keyer_speed;      //100;
+    ps.dit_time         = 1920/ts.keyer_speed;      //100;
 }
```

Comment:
Hi,
appreciate the contribution! And after taking out my calculator I figured 1650 is not the right value (too short pulse). However, 1920 isn't either (too long in theory). It should be 1800: 1500 ticks/s * 1,2 = 1800 = 1WPM. I can see how you calculated 1920 (1650 * 100/86). Have you measured other WPMs (especially slow ones)? And also the pause should be checked, i.e.  a dit and a pause together should be 200ms at 12WPM.  The slight difference between 1920 and 1800 could be due to the "rounding of the edges" the code is doing.